packages/calcite-components/src/components/input-number/input-number.tsx

```diff
@@ -871,6 +871,26 @@ export class InputNumber
     const validNewValue = ["-", "."].includes(newValue) ? "" : newValue;
     this.value = validNewValue;
 
+    const localizedCharWhitelist = new Set([
+      "e",
+      "E",
+      numberStringFormatter.decimal,
+      numberStringFormatter.minusSign,
+      numberStringFormatter.group,
+      ...numberStringFormatter.digits,
+    ]);
+
+    // remove invalid characters from child input
+    if (this?.childNumberEl?.value) {
+      const sanitizedChildElValue = Array.from(this.childNumberEl.value)
```

Comment:
nitpick: since `this.childNumberEl.value` is referenced a few times it might be nice to make it a const.